packages/pds/src/api/com/atproto/sync/getBlocks.ts

```diff
@@ -0,0 +1,25 @@
+import { CID } from 'multiformats/cid'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../../../lexicon'
+import SqlRepoStorage from '../../../../sql-repo-storage'
+import AppContext from '../../../../context'
+import { blocksToCar } from '@atproto/repo'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.sync.getBlocks(async ({ params }) => {
+    const { did } = params
+    const cids = params.cids.map((c) => CID.parse(c))
+    const storage = new SqlRepoStorage(ctx.db, did)
+    const got = await storage.getBlocks(cids)
+    if (got.missing.length > 0) {
+      const missingStr = got.missing.map((c) => c.toString())
+      throw new InvalidRequestError(`Could not find cids: ${missingStr}`)
```

Comment:
Just asking for good measure—would there be any value in exposing `missing` in the output rather than making it an error?

Reply:
Ah, I guess there's not really a good place for it, is there?

Reply:
yeah i thought about that 🤔 

I'm not opposed to just _not_ returning them & making the client figure it out. 

My main thought is that this route should be used for relatively few cids at a time & those cids are like such that if you don't have them _all_ you can't do whatever you were trying to do